keps/sig-storage/5040-remove-gitrepo-driver/README.md

```diff
@@ -0,0 +1,908 @@
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+Since kubelet runs as root an attacker can configure a Git repo to execute
+attacker control code on the host node as root. In [CVE-2024-10220](https://irsl.medium.com/sneaky-write-hook-git-clone-to-root-on-k8s-node-e38236205d54)
+security researchers exploited gitRepo volumes and Git hooks to execute attacker
+controlled code on the host node as root.
+
+gitRepo has been deprecated for almost 5 years and the kubernetes 
+[documentation](https://kubernetes.io/docs/concepts/storage/volumes/#gitrepo)
+warns users against using it and provides a VAP policy to block its use. 
+Kubernetes also recommends mounting an emptyDir volume into an init container 
+that clones the repo using Git, then mounting the emptyDir into the Pod's main
+container. The git-sync project (https://github.com/kubernetes/git-sync) is
+designed for exactly this.
+
+
+### Goals
+
+<!--
+List the specific goals of the KEP. What is it trying to achieve? How will we
+know that this has succeeded?
+-->
+
+- Remove in-tree gitRepo volume support.
+
+### Non-Goals
+
+<!--
+What is out of scope for this KEP? Listing non-goals helps to focus discussion
+and make progress.
+-->
+
+## Proposal
+
+Kubernetes drops support for gitRepo volume types by removing the in-tree driver
+that actually mounts the volume in the pod. In this case we will still keep the
+gitRepo volume in the API. This removal is similar to removal of support for
+[glusterfs](https://kubernetes.io/docs/concepts/storage/volumes/#glusterfs)
+which was deprecated in v1.25 and removed in v1.26.
+
+Sig-storage was actually very close to doing this and had even added a notice of
+removal in the release notes for v1.32. But due to concerns raised by sig-arch
+they rolled back the notice. sig-storage provided [this rationale](https://github.com/kubernetes/kubernetes/issues/125983#issuecomment-2522201283) for considering removal.
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+We will add a new feature-gate called `DisableGitRepoVolumeDriver` which will be
```

Comment:
Gates which remove features should be framed in the affirmative.  The gate should be "GitRepoVolumeDriver", starting in state Deprecated and I think we should start it as off by default (but not locked).

This fixes the problem in 33, forcing people to re-enble it if they need  It's aggressive, but I think low usage (anecodotal from GKE there's very few users) and high risk justify it.

If there is real pushback, we can either:
* Start enabled
* Add a field to disable just hooks and rush that to beta, breaking ONLY people who use hooks

Reply:
I think I agree with:
* a coarse gate to disable the volume plugin
* defaulting in 1.33 to disable the volume plugin
* letting it sit in that state for several releases before locking disabled and deleting the gitRepo volume plugin entirely

Reply:
Updated, thanks!